File: src/objects.py
class Node:
    def __init__(self, id_to_clause, var_to_clause_ids, sat_vars_to_assignment, unit_clauses, pure_vars):
        self.id_to_clause = id_to_clause
        self.var_to_clause_ids = var_to_clause_ids
        self.sat_vars_to_assignment = sat_vars_to_assignment
        self.unit_clauses = unit_clauses
        self.pure_vars = pure_vars

File: src/objects.jl
mutable struct Node
    id_to_clause::Dict{Int, Set{Int}}
    var_to_clause_ids::Dict{Int, Set{Int}}
    sat_vars_to_assignment::Dict{Int, Bool}
    unit_clauses::Set{Int}
    pure_vars::Set{Int}
    
    function Node(id_to_clause::Dict{Int, Set{Int}}, 
                  var_to_clause_ids::Dict{Int, Set{Int}},
                  sat_vars_to_assignment::Dict{Int, Bool},
                  unit_clauses::Set{Int},
                  pure_vars::Set{Int})
        new(id_to_clause, var_to_clause_ids, sat_vars_to_assignment, 
            unit_clauses, pure_vars)
    end
end